bigquery-storage/api/BigQueryStorage.Samples/Quickstart.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START bigquerystorage_quickstart]
+
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Google.Cloud.BigQuery.Storage.V1;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using static Google.Cloud.BigQuery.Storage.V1.ReadSession.Types;
+
+public class QuickstartSample
+{
+    public struct Data
+    {
+        public string Name { get; set; }
+        public string State { get; set; }
+        public long Number { get; set; }
+    }
+
+    public async Task<List<Data>> QuickstartAsync(string projectId)
+    {
+        var bigQueryReadClient = BigQueryReadClient.Create();
+
+        CreateReadSessionRequest createReadSessionRequest = new CreateReadSessionRequest
+        {
+            Parent = $"projects/{projectId}",
+            ReadSession = new ReadSession
+            {
+                // This example uses baby name data from the public datasets.
+                Table = "projects/bigquery-public-data/datasets/usa_names/tables/usa_1910_current",
+                DataFormat = DataFormat.Avro,
+                ReadOptions = new TableReadOptions
+                {
+                    // Specify the columns to be projected by adding them to the selected fields.
+                    SelectedFields = { "name", "number", "state" },
+                    RowRestriction = "state = \"WA\""
```

Comment:
Could we restrict to count of 50 maybe o something similar. Just so this doesn't take too long.

Reply:
I don't think we have a way to put a limit on number of rows and this also makes it coherent to its java counter part.